cmd/broker/expiration_test.go

```diff
@@ -0,0 +1,315 @@
+  "service_id": "47c9dcbf-ff30-448e-ab36-d3bad66ba281",
+  "plan_id": "7d55d31d-35ae-4438-bf13-6ffdfa107d9f"
+}`
+
+func TestExpiration(t *testing.T) {
+	// before all
+	suite := NewBrokerSuiteTest(t)
+	defer suite.TearDown()
+
+	t.Run("should expire a trial instance", func(t *testing.T) {
+		// given
+		instanceID := uuid.NewString()
+		resp := suite.CallAPI(http.MethodPut,
+			fmt.Sprintf(provisioningRequestPathFormat, instanceID),
+			trialProvisioningRequestBody)
+		assert.Equal(t, http.StatusAccepted, resp.StatusCode)
+
+		provisioningOpID := suite.DecodeOperationID(resp)
+		suite.processProvisioningAndReconcilingByOperationID(provisioningOpID)
+		suite.WaitForOperationState(provisioningOpID, domain.Succeeded)
+
+		// when
+		resp = suite.CallAPI(http.MethodPut,
+			fmt.Sprintf(expirationRequestPathFormat, instanceID),
+			"")
+
+		// then
+		assert.Equal(t, http.StatusAccepted, resp.StatusCode)
+
+		suspensionOpID := suite.DecodeOperationID(resp)
+		assert.NotEmpty(t, suspensionOpID)
+
+		suite.WaitForOperationState(suspensionOpID, domain.InProgress)
+		suite.MarkClusterConfigurationDeleted(instanceID)
+		suite.FinishDeprovisioningOperationByProvisionerForGivenOpId(suspensionOpID)
+		suite.WaitForOperationState(suspensionOpID, domain.Succeeded)
+
+		actualInstance := suite.GetInstance(instanceID)
+		assert.True(t, actualInstance.IsExpired())
+		assert.False(t, *actualInstance.Parameters.ErsContext.Active)
+	})
+
+	t.Run("should retrigger expiration (suspension) on already expired instance", func(t *testing.T) {
+		// given
+		instanceID := uuid.NewString()
+		resp := suite.CallAPI(http.MethodPut,
+			fmt.Sprintf(provisioningRequestPathFormat, instanceID),
+			trialProvisioningRequestBody)
+		assert.Equal(t, http.StatusAccepted, resp.StatusCode)
+
+		provisioningOpID := suite.DecodeOperationID(resp)
+		suite.processProvisioningAndReconcilingByOperationID(provisioningOpID)
+		suite.WaitForOperationState(provisioningOpID, domain.Succeeded)
+
+		// when
+		resp = suite.CallAPI(http.MethodPut,
+			fmt.Sprintf(expirationRequestPathFormat, instanceID),
+			"")
+
+		// then
+		assert.Equal(t, http.StatusAccepted, resp.StatusCode)
+
+		suspensionOpID := suite.DecodeOperationID(resp)
+		assert.NotEmpty(t, suspensionOpID)
+
+		suite.WaitForOperationState(suspensionOpID, domain.InProgress)
+		suite.MarkClusterConfigurationDeleted(instanceID)
+		suite.FinishDeprovisioningOperationByProvisionerForGivenOpId(suspensionOpID)
+		suite.WaitForOperationState(suspensionOpID, domain.Succeeded)
+
+		actualInstance := suite.GetInstance(instanceID)
+		assert.True(t, actualInstance.IsExpired())
+		assert.False(t, *actualInstance.Parameters.ErsContext.Active)
+
+		// when
+		resp = suite.CallAPI(http.MethodPut,
+			fmt.Sprintf(expirationRequestPathFormat, instanceID),
+			"")
+
+		// then
+		assert.Equal(t, http.StatusAccepted, resp.StatusCode)
+
+		suspensionOpID2 := suite.DecodeOperationID(resp)
+		assert.NotEmpty(t, suspensionOpID2)
```

Comment:
Could we assert that suspensionOpID1 != suspensionOpID2?

Reply:
sure, I will add the assertion.